lib/common.h

```diff
@@ -30,8 +30,24 @@
 #include <limits.h>
 #include <stdint.h>
 
+/* Portability */
+#if defined(__FreeBSD__)
```

Comment:
I think this needs one of two things to happen:

1.) Figure out which situation is more common amongst the most platforms (sha.h vs. sha1.h and needing SHA1_* macros) and making that the default, or

2.) Throw in an #else #error "Unsure of what to do for inclusion of sha1 bits in this environment" to give a friendly pointer as to why this doesn't work when a different OS is attempted rather than compilation errors later on at first use.

I think I'm largely indifferent to the exact approach.

Reply:
Picked option 2 because I'm not going to go look.